diff --git a/wangle/acceptor/test/AcceptorTest.cpp b/wangle/acceptor/test/AcceptorTest.cpp
index 20f67bb..a864720 100644
--- a/wangle/acceptor/test/AcceptorTest.cpp
+++ b/wangle/acceptor/test/AcceptorTest.cpp
@@ -13,10 +13,13 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include <folly/logging/Init.h>
 #include <wangle/acceptor/Acceptor.h>
 
 #include <gtest/gtest.h>
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using namespace folly;
 using namespace testing;
 
diff --git a/wangle/acceptor/test/ConnectionManagerTest.cpp b/wangle/acceptor/test/ConnectionManagerTest.cpp
index 2d36fa4..fa9042f 100644
--- a/wangle/acceptor/test/ConnectionManagerTest.cpp
+++ b/wangle/acceptor/test/ConnectionManagerTest.cpp
@@ -13,12 +13,15 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include <folly/logging/Init.h>
 #include <wangle/acceptor/ConnectionManager.h>
 
 #include <gtest/gtest.h>
 #include <gmock/gmock.h>
 #include <gflags/gflags.h>
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using namespace folly;
 using namespace testing;
 using namespace wangle;
diff --git a/wangle/acceptor/test/LoadShedConfigurationTest.cpp b/wangle/acceptor/test/LoadShedConfigurationTest.cpp
index 446369e..772c61a 100644
--- a/wangle/acceptor/test/LoadShedConfigurationTest.cpp
+++ b/wangle/acceptor/test/LoadShedConfigurationTest.cpp
@@ -13,10 +13,13 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include <folly/logging/Init.h>
 #include <wangle/acceptor/LoadShedConfiguration.h>
 
 #include <gtest/gtest.h>
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using namespace wangle;
 using namespace testing;
 
diff --git a/wangle/acceptor/test/PeekingAcceptorHandshakeHelperTest.cpp b/wangle/acceptor/test/PeekingAcceptorHandshakeHelperTest.cpp
index 122cbde..ed39b75 100644
--- a/wangle/acceptor/test/PeekingAcceptorHandshakeHelperTest.cpp
+++ b/wangle/acceptor/test/PeekingAcceptorHandshakeHelperTest.cpp
@@ -18,10 +18,13 @@
 #include <thread>
 
 #include <folly/io/async/test/MockAsyncSSLSocket.h>
+#include <folly/logging/Init.h>
 #include <folly/portability/GMock.h>
 #include <folly/portability/GTest.h>
 #include <wangle/acceptor/test/AcceptorHelperMocks.h>
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using namespace folly;
 using namespace folly::test;
 using namespace wangle;
diff --git a/wangle/bootstrap/test/BootstrapTest.cpp b/wangle/bootstrap/test/BootstrapTest.cpp
index 1281cb9..1f9afc2 100644
--- a/wangle/bootstrap/test/BootstrapTest.cpp
+++ b/wangle/bootstrap/test/BootstrapTest.cpp
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+#include <folly/logging/Init.h>
 #include "wangle/bootstrap/ServerBootstrap.h"
 #include "wangle/bootstrap/ClientBootstrap.h"
 #include "wangle/channel/Handler.h"
@@ -24,6 +25,8 @@
 #include <folly/String.h>
 #include <folly/experimental/TestUtil.h>
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using namespace wangle;
 using namespace folly;
 
diff --git a/wangle/channel/broadcast/test/BroadcastHandlerTest.cpp b/wangle/channel/broadcast/test/BroadcastHandlerTest.cpp
index 09b7f10..d8c88fe 100644
--- a/wangle/channel/broadcast/test/BroadcastHandlerTest.cpp
+++ b/wangle/channel/broadcast/test/BroadcastHandlerTest.cpp
@@ -13,9 +13,12 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include <folly/logging/Init.h>
 #include <wangle/channel/broadcast/test/Mocks.h>
 #include <wangle/channel/test/MockHandler.h>
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using namespace wangle;
 using namespace folly;
 using namespace testing;
diff --git a/wangle/channel/broadcast/test/BroadcastPoolTest.cpp b/wangle/channel/broadcast/test/BroadcastPoolTest.cpp
index dbc2e7f..59404ca 100644
--- a/wangle/channel/broadcast/test/BroadcastPoolTest.cpp
+++ b/wangle/channel/broadcast/test/BroadcastPoolTest.cpp
@@ -13,12 +13,15 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include <folly/logging/Init.h>
 #include <wangle/bootstrap/ServerBootstrap.h>
 #include <wangle/channel/broadcast/BroadcastPool.h>
 #include <wangle/channel/broadcast/test/Mocks.h>
 
 #include "wangle/channel/Handler.h"
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using namespace folly;
 using namespace testing;
 using namespace wangle;
diff --git a/wangle/channel/broadcast/test/ObservingHandlerTest.cpp b/wangle/channel/broadcast/test/ObservingHandlerTest.cpp
index d76e8ff..57434f9 100644
--- a/wangle/channel/broadcast/test/ObservingHandlerTest.cpp
+++ b/wangle/channel/broadcast/test/ObservingHandlerTest.cpp
@@ -13,10 +13,13 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include <folly/logging/Init.h>
 #include <wangle/channel/broadcast/test/Mocks.h>
 #include <wangle/channel/test/MockHandler.h>
 #include <wangle/codec/MessageToByteEncoder.h>
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using namespace wangle;
 using namespace folly;
 using namespace testing;
diff --git a/wangle/channel/test/AsyncSocketHandlerTest.cpp b/wangle/channel/test/AsyncSocketHandlerTest.cpp
index 79c4c2c..b5fb924 100644
--- a/wangle/channel/test/AsyncSocketHandlerTest.cpp
+++ b/wangle/channel/test/AsyncSocketHandlerTest.cpp
@@ -14,12 +14,15 @@
  * limitations under the License.
  */
 
+#include <folly/logging/Init.h>
 #include <gtest/gtest.h>
 #include <wangle/channel/AsyncSocketHandler.h>
 #include <wangle/channel/Pipeline.h>
 #include <wangle/channel/test/MockHandler.h>
 #include <wangle/channel/test/MockPipeline.h>
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using namespace folly;
 using namespace testing;
 using namespace wangle;
diff --git a/wangle/channel/test/OutputBufferingHandlerTest.cpp b/wangle/channel/test/OutputBufferingHandlerTest.cpp
index 2a44290..a1635fd 100644
--- a/wangle/channel/test/OutputBufferingHandlerTest.cpp
+++ b/wangle/channel/test/OutputBufferingHandlerTest.cpp
@@ -18,9 +18,12 @@
 #include <wangle/channel/OutputBufferingHandler.h>
 #include <wangle/channel/test/MockHandler.h>
 #include <folly/io/async/AsyncSocket.h>
+#include <folly/logging/Init.h>
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using namespace folly;
 using namespace wangle;
 using namespace testing;
diff --git a/wangle/channel/test/PipelineTest.cpp b/wangle/channel/test/PipelineTest.cpp
index 4562985..3253641 100644
--- a/wangle/channel/test/PipelineTest.cpp
+++ b/wangle/channel/test/PipelineTest.cpp
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+#include <folly/logging/Init.h>
 #include <wangle/channel/Handler.h>
 #include <wangle/channel/Pipeline.h>
 #include <wangle/channel/StaticPipeline.h>
@@ -24,6 +25,8 @@
 #include <gtest/gtest.h>
 #include <boost/thread/barrier.hpp>
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using namespace folly;
 using namespace wangle;
 using namespace testing;
diff --git a/wangle/client/ssl/SSLSessionCallbacks.cpp b/wangle/client/ssl/SSLSessionCallbacks.cpp
index f46d9bc..14eca00 100644
--- a/wangle/client/ssl/SSLSessionCallbacks.cpp
+++ b/wangle/client/ssl/SSLSessionCallbacks.cpp
@@ -59,7 +59,7 @@ SSLSessionCallbacks* SSLSessionCallbacks::getCacheFromContext(SSL_CTX* ctx) {
 // static
 std::string SSLSessionCallbacks::getSessionKeyFromSSL(SSL* ssl) {
   auto sock = folly::AsyncSSLSocket::getFromSSL(ssl);
-  return sock ? sock->getSessionKey() : "";
+  return sock ? sock->getServiceIdentity() : "";
 }
 
 // static
diff --git a/wangle/cmake/FindFolly.cmake b/wangle/cmake/FindFolly.cmake
index 93708af..38840b1 100644
--- a/wangle/cmake/FindFolly.cmake
+++ b/wangle/cmake/FindFolly.cmake
@@ -16,9 +16,10 @@ CMAKE_MINIMUM_REQUIRED(VERSION 2.8.7 FATAL_ERROR)
 INCLUDE(FindPackageHandleStandardArgs)
 
 FIND_LIBRARY(FOLLY_LIBRARY folly PATHS ${FOLLY_LIBRARYDIR})
+FIND_LIBRARY(FOLLY_LOGGING_LIBRARY follylogging PATHS ${FOLLY_LIBRARYDIR})
 FIND_PATH(FOLLY_INCLUDE_DIR "folly/String.h" PATHS ${FOLLY_INCLUDEDIR})
 
-SET(FOLLY_LIBRARIES ${FOLLY_LIBRARY})
+SET(FOLLY_LIBRARIES ${FOLLY_LIBRARY} ${FOLLY_LOGGING_LIBRARY})
 
 FIND_PACKAGE_HANDLE_STANDARD_ARGS(Folly
   REQUIRED_ARGS FOLLY_INCLUDE_DIR FOLLY_LIBRARIES)
diff --git a/wangle/codec/test/CodecTest.cpp b/wangle/codec/test/CodecTest.cpp
index 3698601..bdc6781 100644
--- a/wangle/codec/test/CodecTest.cpp
+++ b/wangle/codec/test/CodecTest.cpp
@@ -15,6 +15,7 @@
  */
 
 #include <gtest/gtest.h>
+#include <folly/logging/Init.h>
 
 #include <wangle/codec/FixedLengthFrameDecoder.h>
 #include <wangle/codec/LengthFieldBasedFrameDecoder.h>
@@ -22,6 +23,8 @@
 #include <wangle/codec/LineBasedFrameDecoder.h>
 #include <wangle/codec/test/CodecTestUtils.h>
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using namespace folly;
 using namespace wangle;
 using namespace folly::io;
diff --git a/wangle/deprecated/rx/test/RxTest.cpp b/wangle/deprecated/rx/test/RxTest.cpp
index 6174557..dea69d0 100644
--- a/wangle/deprecated/rx/test/RxTest.cpp
+++ b/wangle/deprecated/rx/test/RxTest.cpp
@@ -14,10 +14,13 @@
  * limitations under the License.
  */
 
+#include <folly/logging/Init.h>
 #include <wangle/deprecated/rx/Observer.h>
 #include <wangle/deprecated/rx/Subject.h>
 #include <gtest/gtest.h>
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using namespace wangle;
 
 static std::unique_ptr<Observer<int>> incrementer(int& counter) {
diff --git a/wangle/ssl/test/SSLContextManagerTest.cpp b/wangle/ssl/test/SSLContextManagerTest.cpp
index 6c31d43..618f78e 100644
--- a/wangle/ssl/test/SSLContextManagerTest.cpp
+++ b/wangle/ssl/test/SSLContextManagerTest.cpp
@@ -15,12 +15,15 @@
  */
 #include <folly/io/async/EventBase.h>
 #include <folly/io/async/SSLContext.h>
+#include <folly/logging/Init.h>
 #include <glog/logging.h>
 #include <gtest/gtest.h>
 #include <wangle/ssl/SSLCacheOptions.h>
 #include <wangle/ssl/SSLContextManager.h>
 #include <wangle/acceptor/SSLContextSelectionMisc.h>
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using std::shared_ptr;
 using namespace folly;
 
diff --git a/wangle/ssl/test/TLSCredProcessorTest.cpp b/wangle/ssl/test/TLSCredProcessorTest.cpp
index 288b115..643321b 100644
--- a/wangle/ssl/test/TLSCredProcessorTest.cpp
+++ b/wangle/ssl/test/TLSCredProcessorTest.cpp
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 #include <boost/filesystem.hpp>
+#include <folly/logging/Init.h>
 #include <folly/portability/GTest.h>
 #include <folly/synchronization/Baton.h>
 #include <folly/File.h>
@@ -22,6 +23,8 @@
 #include <wangle/ssl/TLSCredProcessor.h>
 #include <wangle/ssl/test/TicketUtil.h>
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using namespace testing;
 using namespace folly;
 using namespace wangle;
diff --git a/wangle/util/test/FilePollerTest.cpp b/wangle/util/test/FilePollerTest.cpp
index 1102b60..c248fbc 100644
--- a/wangle/util/test/FilePollerTest.cpp
+++ b/wangle/util/test/FilePollerTest.cpp
@@ -18,6 +18,7 @@
 #include <mutex>
 
 #include <boost/filesystem.hpp>
+#include <folly/logging/Init.h>
 #include <folly/portability/GTest.h>
 #include <folly/synchronization/Baton.h>
 #include <folly/File.h>
@@ -25,6 +26,8 @@
 #include <folly/experimental/TestUtil.h>
 #include <wangle/util/FilePoller.h>
 
+FOLLY_INIT_LOGGING_CONFIG(".=WARNING,folly=INFO; default:async=true");
+
 using namespace testing;
 using namespace folly;
 using namespace wangle;
